code/renderers/vue/src/docs/extractArgTypes.ts

```diff
@@ -11,7 +11,7 @@ const SECTIONS = ['props', 'events', 'slots', 'methods'];
 function isEnum(propDef: PropDef, docgenInfo: DocgenInfo): false | PropDef {
   // cast as any, since "values" doesn't exist in DocgenInfo type
   const { type, values } = docgenInfo as any;
-  const matched = Array.isArray(values) && values.length && type?.name === 'enum';
+  const matched = Array.isArray(values) && values.length && type && type.name !== 'enum';
```

Comment:
```suggestion
  const matched = Array.isArray(values) && values.length && type && type.name === 'enum';
```

This should fix your issue, while leaving my original fix intact

Reply:
In my original PR, that equality check was still wrong, it just didn't surface until the move to optional chaining

Reply:
I probably misunderstood your question
> Does that second render only work like that when the equality is reversed, or does it also work this way once the type check is done separately instead of chaining?

The second render works in the following cases:
```ts
const matched = Array.isArray(values) && values.length && type?.name !== 'enum';
```
```ts
const matched = Array.isArray(values) && values.length && type && type.name !== 'enum';
```
but not in case the equality is not reversed:
```ts
const matched = Array.isArray(values) && values.length && type && type.name === 'enum';
```

Reply:
Oh ok, hmm... I'll have to take another look at this

Reply:
Hey @ccatterina , sorry for the late reply - Covid finally made it into our home, so I've been super busy with a sick and preggo wife, and a sick toddler.

If I'm understanding you correctly, the neg type check is so that we're casting anything that _behaves_ like an `enum` into one? I wonder if adding a small docblock would help clear this up in the future.

I dug into some of these other resources, and it looks like there's specifically an `@enum` directive that isn't implemented in vuedocgen.

@valentinpalkovic I think I'm going to resolve this conversation as I think the proposed solution is the way.

Reply:
Given that you two (seemingly smart people!) have discussed how this single character make it all behave, I think it makes sense to refactor the code to be more readable for future work. Either by making the `matched` variable more verbose, or by adding comments describing what the conditions do, and why.

I don't know much about Vue so I trust that you two have got it under control, but the next person is probably going to be just as confused as you initially.

Reply:
I agree, I think that the most confusing thing is the name of the function `isEnum`, maybe something like `inferEnum` comes out clearer.

I could rename the function and a comment to describe its purpose.

Reply:
@JReinhold @ProjektGopher I pushed a new commit that tries to make the purpose of the function clearer, feel free to make some changes (my english is not so good).